carcara/src/checker/rules/strings.rs

```diff
@@ -57,6 +57,52 @@ fn is_prefix(
     Ok(p_flat)
 }
 
+type Suffixes = (Vec<Rc<Term>>, Vec<Rc<Term>>);
+
+fn strip_prefix(
+    s: Rc<Term>,
+    t: Rc<Term>,
+    pool: &mut dyn TermPool,
+    rev: bool,
+    polyeq_time: &mut Duration,
+) -> Result<Suffixes, CheckerError> {
+    let mut s_flat = flatten(s.clone(), pool);
+    let mut t_flat = flatten(t.clone(), pool);
+
+    if rev {
+        s_flat.reverse();
+        t_flat.reverse();
+    }
+
+    if s_flat.is_empty() {
+        return Err(CheckerError::ExpectedConcatApplication(s));
+    }
+    if t_flat.is_empty() {
+        return Err(CheckerError::ExpectedConcatApplication(t));
+    }
```

Comment:
Instead of creating a new error variant, you can use `TermOfWrongForm` (this is the error variant returned by `match_term_err!`)
```suggestion
    if s_flat.is_empty() {
        return Err(CheckerError::TermOfWrongForm("(str.++ ...)", s));
    }
    if t_flat.is_empty() {
        return Err(CheckerError::TermOfWrongForm("(str.++ ...)", t));
    }
```